### PR TITLE
Make storage initializers required, just like local variables

### PR DESCRIPTION
**Pull request by mohammadfawaz**
Closes https://github.com/FuelLabs/sway/issues/2238

- Make storage initializers required
- Update all the tests and examples
- Improve the docs for storage variables
image

### CODE REVIEW

File: docs/src/blockchain-development/storage.md [hunk 1]
```diff
@@ -8,33 +8,23 @@ Some basic use cases of storage include declaring an owner address for a contrac
...
 ```sway
-storage.var1 = v;
+{{#include ../../../examples/storage_variables/src/main.sw:storage_write}}
 ```
 
 To read a storage variable, you also need to use the `storage` keyword as follows:
 
 ```sway
-let v = storage.var1;
+{{#include ../../../examples/storage_variables/src/main.sw:storage_read}}
```
Comment (sezna) [inline docs/src/blockchain-development/storage.md:25]: good catch

File: examples/storage_map/src/main.sw [hunk 2]
```diff
@@ -9,8 +9,11 @@ struct Data {
 
 storage {
     map1: StorageMap<u64,
-    u64>, map2: StorageMap<(b256,
-    bool), Data>, 
+    u64> = StorageMap {
```
Comment (sezna) [inline examples/storage_map/src/main.sw:12]: @eureka-cpu wen formatter

**Review by sezna — APPROVED** [review]
Looks good to me. To summarize, if I understand correctly:
1. Update parser to remove optionality around initializer 
2. remove any optionality in the type checker and constant evaluation of initializers
3. add initializers to all examples which were missing them

No new logic seems to be added, so this seems safe -- just removing the option stuff. 

Nice PR, very thorough with all the examples and sway fmt stuff. 💪🏻

**Comment by mohammadfawaz** [comment]
> Looks good to me. To summarize, if I understand correctly:
> 
> 1. Update parser to remove optionality around initializer
> 2. remove any optionality in the type checker and constant evaluation of initializers
> 3. add initializers to all examples which were missing them
> 
> No new logic seems to be added, so this seems safe -- just removing the option stuff.
> 
> Nice PR, very thorough with all the examples and sway fmt stuff. 💪🏻

That's a good summary. As for the formatter stuff, `sway-fmt-v2` seems to be dropping initializers actually so I filed in an issue here: https://github.com/FuelLabs/sway/issues/2279

File: examples/storage_map/src/main.sw (hunk 2 above)
Comment (eureka-cpu) [reply examples/storage_map/src/main.sw:12]: soon 😇

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**